pkg/handler/data_recorder_kafka.go

```diff
@@ -36,11 +36,19 @@ var NewKafkaRecorder = func() DataRecorder {
 		config.Config.RecorderKafkaKeyFile,
 		config.Config.RecorderKafkaCAFile,
 		config.Config.RecorderKafkaVerifySSL,
+		config.Config.RecorderKafkaSimpleSSL,
 	)
 	if tlscfg != nil {
 		cfg.Net.TLS.Enable = true
 		cfg.Net.TLS.Config = tlscfg
 	}
+
+	if config.Config.RecorderKafkaUsername != "" && config.Config.RecorderKafkaPassword != "" {
+		cfg.Net.SASL.Enable = true
+	}
+	cfg.Net.SASL.User = config.Config.RecorderKafkaUsername
+	cfg.Net.SASL.Password = config.Config.RecorderKafkaPassword
```

Comment:
I think you can also move these two lines into the `if` block, so that if one of them is empty/non-empty, we can minimize the unexpected side effect to the existing kafka config.